### PR TITLE
update guesser and processor for ITE6 OSV Certifier attestation

### PR DESCRIPTION
**Pull request by pxp928**
Signed-off-by: pxp928 <parth.psu@gmail.com>

Part of issue https://github.com/guacsec/guac/issues/222

### CODE REVIEW

**Comment by pxp928** [comment]
The format of the attestation is still open for discussion and can me changed/modified based on feedback.

**Review by mihaimaruseac — APPROVED** [review]

File: pkg/handler/processor/processor.go [hunk 1]
```diff
@@ -54,6 +54,7 @@ type DocumentType string
 const (
 	DocumentITE6SLSA    DocumentType = "SLSA"
 	DocumentITE6Generic DocumentType = "ITE6"
+	DocumentITE6OSV     DocumentType = "ITE6OSV"
```
Comment (lumjjb) [inline pkg/handler/processor/processor.go:57]: should this be ITE6SCAI instead?

**Comment by pxp928** [comment]
I will be updating this PR based on some upstream changes to the SCAI predicate.

**Comment by pxp928** [comment]
Updated the Predicate. @lumjjb @mlieberman85 Please take a look and provide feedback if it's generic enough. Based off -> https://github.com/sigstore/cosign/blob/main/specs/COSIGN_VULN_ATTESTATION_SPEC.md

File: pkg/handler/processor/guesser/type_ite6.go [hunk 2]
```diff
@@ -35,6 +35,8 @@ func (_ *ite6TypeGuesser) GuessDocumentType(blob []byte, format processor.Format
 				return processor.DocumentITE6Generic
 			} else if strings.HasPrefix(statement.PredicateType, "https://in-toto.io/attestation/certify/v0.1") {
 				return processor.DocumentITE6Generic
+			} else if strings.HasPrefix(statement.PredicateType, "http://in-toto.io/attestation/vuln/v0.1") {
```
Comment (mlieberman85) [inline pkg/handler/processor/guesser/type_ite6.go:38]: Do we want to be consistent with http vs. https? I don't think it really matters, but not sure.

**Review by mlieberman85 — APPROVED** [review]
LGTM. I think any changes we need to make will become apparent once we start using this more in earnest.

**Pull request merged**